src/devices/Mpu9250/Mpu9250.cs

```diff
@@ -0,0 +1,996 @@
+            // ACCEL_CONFIG 2 (0x1D) set ACCEL_FCHOICE_B = 0 and A_DLPFCFG[2:0]=1(b001)
+            // Bandwidth for Accelerometer to 184Hz
+            AccelerometerBandwidth = AccelerometerBandwidth.Bandwidth0184Hz;
+            // Enable Motion Interrupt
+            //  In INT_ENABLE (0x38), set the whole register to 0x40 to enable motion interrupt only
+            WriteRegister(Register.INT_ENABLE, 0x40);
+            // Enable AccelHardware Intelligence:
+            // In MOT_DETECT_CTRL (0x69), set ACCEL_INTEL_EN = 1 and ACCEL_INTEL_MODE  = 1
+            WriteRegister(Register.MOT_DETECT_CTRL, 0b1100_0000);
+            // Set Motion Threshold:
+            // In WOM_THR (0x1F), set the WOM_Threshold[7:0] to 1~255 LSBs (0~1020mg)
+            WriteRegister(Register.WOM_THR, (byte)magnetometerThreshold);
+            // Set Frequency of Wake-up:
+            // In LP_ACCEL_ODR (0x1E), set Lposc_clksel[3:0] = 0.24Hz ~ 500Hz
+            WriteRegister(Register.LP_ACCEL_ODR, (byte)accelerometerLowPowerFrequency);
+            // Enable Cycle Mode (AccelLow Power Mode):
+            // In PWR_MGMT_1 (0x6B) make CYCLE =1
+            WriteRegister(Register.PWR_MGMT_1, 0b0010_0000);
+            // Motion Interrupt Configuration Completed
+        }
+
+        /// <summary>
+        /// Initialize the MPU9250
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device</param>
+        /// <param name="autoDispose">Will automatically dispose the I2C device if true</param>
+        public Mpu9250(I2cDevice i2cDevice, bool autoDispose = true)
+        {
+            _i2cDevice = i2cDevice;
+            _autoDispose = autoDispose;
+            Reset();
+            PowerOn();
+            if (!CheckVersion())
+                throw new IOException($"This device does not contain the correct signature 0x71 for a MPU9250");
+            GyroscopeBandwidth = GyroscopeBandwidth.Bandwidth0250Hz;
+            GyroscopeRange = GyroscopeRange.Range0250Dps;
+            AccelerometerBandwidth = AccelerometerBandwidth.Bandwidth1130Hz;
+            AccelerometerRange = AccelerometerRange.Range02G;
+            // Setup I2C for the AK8963
+            WriteRegister(Register.USER_CTRL, (byte)UserControls.I2C_MST_EN);
+            // Speed of 400 kHz
+            WriteRegister(Register.I2C_MST_CTRL, (byte)I2cBussFrequency.Freqency400kHz);
+            _ak8963 = new Ak8963.Ak8963(_i2cDevice, new Ak8963Attached(), false);
+            Thread.Sleep(50);
+            if (!_ak8963.CheckVersion())
+            {
+                // Try to reset the device first
+                _ak8963.Reset();
+                // Wait a bit
+                if (!_ak8963.CheckVersion())
+                {
+                    // Try to reset the I2C Bus
+                    WriteRegister(Register.USER_CTRL, (byte)UserControls.I2C_MST_RST);
+                    Thread.Sleep(100);
```

Comment:
please explain sleeps

Reply:
Poorly documented time to wait after the I2C bus reset. Found out that waiting a little bit is needed. Exact time may be lower. So added this.